src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,126 @@
+package duke;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.ListCommand;
+import duke.command.LoadCommand;
+import duke.command.MarkCommand;
+import duke.command.MatchCommand;
+import duke.command.SaveCommand;
+import duke.command.UnmarkCommand;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+/**
+ * Deals with making sense of the user command.
+ */
+public class Parser {
+    /**
+     * Parses the user command and returns the corresponding command object.
+     *
+     * @param command The user command
+     * @return The corresponding command object
+     * @throws DukeException if the command is invalid
+     */
+    public static Command parse(String command) throws DukeException {
+        switch (command) {
```

Comment:
Good that you used switch statements instead of if-else to make the code more readable